core/src/main/scala/cr/pulsar/Reader.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 Chatroulette
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cr.pulsar
+
+import cats._
+import cats.effect._
+import cats.implicits._
+import cr.pulsar.internal.FutureLift._
+import fs2._
+import org.apache.pulsar.client.api.{ Message, MessageId }
+import scala.util.control.NoStackTrace
+
+/**
+ * A Reader can be used to read all the messages currently available in a topic.
+ */
+trait Reader[F[_]] {
+  def read: Stream[F, Message[Array[Byte]]]
```

Comment:
I would have expected the interface to take in a `MessageId`, no?

```scala
def read(id: MessageId): Stream[F, Message[Array[Byte]]]
```

Or do we have a different use case for this?

Reply:
You are right. We should be able to pass `messageId`, but it's part of the constructor of the reader rather than the method argument.